Security Session not compatible with native image

The class https://github.com/micronaut-projects/micronaut-security/blob/dbabb49615cf701e67b7b6868207d77a3a52f1b9/security/src/main/java/io/micronaut/security/authentication/AuthenticationUserDetailsAdapter.java is placed into the session by `micronaut-security-session` but has no `@Introspected` or reflection annotations so breaks when running in native with:

```
Caused by: com.fasterxml.jackson.databind.exc.InvalidDefinitionException: No serializer found for class io.micronaut.security.authentication.AuthenticationUserDetailsAdapter and no properties discovered to create BeanSerializer (to avoid exception, disable SerializationFeature.FAIL_ON_EMPTY_BEANS)
	at com.fasterxml.jackson.databind.SerializerProvider.reportBadDefinition(SerializerProvider.java:1276)
	at com.fasterxml.jackson.databind.DatabindContext.reportBadDefinition(DatabindContext.java:400)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.failForEmpty(UnknownSerializer.java:71)
	at com.fasterxml.jackson.databind.ser.impl.UnknownSerializer.serialize(UnknownSerializer.java:33)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:480)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:319)
	a
```

See https://github.com/pgressa/oraclecloud-cloudnative/issues/86

--- a/security/src/main/java/io/micronaut/security/authentication/AuthenticationUserDetailsAdapter.java
+++ b/security/src/main/java/io/micronaut/security/authentication/AuthenticationUserDetailsAdapter.java
@@ -15,6 +15,7 @@
  */
 package io.micronaut.security.authentication;
 
+import io.micronaut.core.annotation.Introspected;
 import io.micronaut.core.annotation.NonNull;
 import java.util.Map;
 
@@ -24,6 +25,7 @@ import java.util.Map;
  * @author Sergio del Amo
  * @since 1.0
  */
+@Introspected
 public class AuthenticationUserDetailsAdapter implements Authentication {
 
     private final UserDetails userDetails;

